src/Features/Core/Portable/EmbeddedLanguages/DateAndTime/EmbeddedCompletionContext.cs

```diff
@@ -33,31 +52,91 @@ internal partial class DateAndTimeEmbeddedCompletionProvider
 
-            private static DateTime s_exampleDateTime = DateTime.Parse("2009-06-15T13:45:30.1234567Z");
-            private static CultureInfo s_enUsCulture = CultureInfo.GetCultureInfo("en-US");
-            private static CultureInfo s_primaryCulture = CultureInfo.CurrentCulture;
-            private static CultureInfo? s_secondaryCulture = s_primaryCulture.Equals(s_enUsCulture) ? null : s_enUsCulture;
+            private static (string prefix, string suffix) GetUserFormatParts(
+                VirtualCharSequence virtualChars, int startPosition, int endPosition)
+            {
+                virtualChars = virtualChars.IsDefault ? VirtualCharSequence.Empty : virtualChars;
 
-            public void AddStandard(string displayText, string suffix, string description)
-                => Add(displayText, suffix, description, standard: true);
+                using var _1 = PooledStringBuilder.GetInstance(out var prefix);
+                using var _2 = PooledStringBuilder.GetInstance(out var suffix);
+                foreach (var ch in virtualChars)
+                {
+                    if (ch.Span.End <= startPosition)
+                        ch.AppendTo(prefix);
+                    else if (ch.Span.Start >= endPosition)
+                        ch.AppendTo(suffix);
+                }
 
-            public void AddCustom(string displayText, string suffix, string description)
-                => Add(displayText, suffix, description, standard: false);
+                return (prefix.ToString(), suffix.ToString());
+            }
 
-            private void Add(string displayText, string suffix, string description, bool standard)
+            private void AddExamples(ArrayBuilder<string> examples, bool standard, string displayText)
+            {
+                var userFormat = _userFormatPrefix + displayText + _userFormatSuffix;
+
+                // The total user format may not be legal.  So try to format the date, but be tolerant of that failing.
+                TryAddExample(examples, standard, userFormat, s_primaryCulture);
+                TryAddExample(examples, standard, userFormat, s_secondaryCulture);
+
+                // We should never fail to produce an example string for one of the format strings we are presenting.
+                AddExample(examples, standard, displayText, s_primaryCulture);
+                AddExample(examples, standard, displayText, s_secondaryCulture);
+            }
+
+            private void TryAddExample(ArrayBuilder<string> examples, bool standard, string displayText, CultureInfo culture)
             {
-                using var _ = PooledStringBuilder.GetInstance(out var descriptionBuilder);
+                try
+                {
+                    AddExample(examples, standard, displayText, culture);
+                }
+                catch (FormatException)
+                {
+                }
+                catch (ArgumentOutOfRangeException)
+                {
+                }
```

Comment:
Can we surround just the ToString() call rather than the other string manipulation bit? Only so we're not catching unrelated exceptions?

Reply:
sure.